audio/out/ao_wasapi.c

```diff
@@ -220,18 +222,25 @@ static DWORD __stdcall AudioThread(void *lpParameter)
 
     MP_DBG(ao, "Entering dispatch loop\n");
     while (true) {
-        if (WaitForSingleObject(state->hWake, INFINITE) != WAIT_OBJECT_0)
-            MP_ERR(ao, "Unexpected return value from WaitForSingleObject\n");
-
-        int thread_state = atomic_load(&state->thread_state);
-        switch (thread_state) {
-        case WASAPI_THREAD_FEED:
+        HANDLE handles[] = {state->hWake, state->hUserWake};
+        switch (WaitForMultipleObjects(MP_ARRAY_SIZE(handles), handles, FALSE, INFINITE)) {
+        case WAIT_OBJECT_0:
```

Comment:
There is one possible issue with this. If multiple objects become signaled, the function returns the index of the first handle in the array whose object was signaled. So there is possibility that we ignore dispatch wake up if both happen.